src/particle/converters/bimap.py

```diff
@@ -18,9 +18,29 @@
 from .. import data
 from ..exceptions import MatchingIDNotFound
 
-
-class BiMap(object):
+from typing import (
+    Union,
+    TextIO,
+    Callable,
+    TypeVar,
+    Generic,
+    overload,
```

Comment:
For looks, maybe better to put this decorator at the end of the import, since not quite the same as the other imports?

Reply:
Sure, I can sort these, which will do it.